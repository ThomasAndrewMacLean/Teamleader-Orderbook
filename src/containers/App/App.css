

.App-header {
  background-color:var(--color-mint);
  height: 60px;
  line-height: 20px;
  font-size: 20px;
  color: white;
  font-weight: 100;
}


.App{
  position: relative;
  margin: auto;
  max-width: 800px;
  height: 100vh;
  /* height: 120%; */
  background: white;
}

.username{
  display: inline;
  /* font-size: 13px; */
  margin-right: -40px;
}

.orderList{
  clear: both;
}



