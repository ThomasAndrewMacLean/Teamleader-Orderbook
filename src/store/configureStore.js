import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from './../reducers/reducers';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { routerMiddleware } from 'react-router-redux';

import createHistory from 'history/createBrowserHistory';



const logger = createLogger({
    collapsed: true
});

const history = createHistory();
const routerMw = routerMiddleware(history);

const composeEnhancers =
    typeof window === 'object' &&
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
            // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
        }) : compose;


export default function configureStore() {
    return createStore(

        rootReducer,
        composeEnhancers(
            applyMiddleware(routerMw, logger, thunk))
    );
}
