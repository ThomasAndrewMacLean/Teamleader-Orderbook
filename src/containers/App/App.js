import React, { Component } from 'react';
import { connect } from 'react-redux';
import List from './../List/List';
import PropTypes from 'prop-types';
import { addOrder } from './../../actions/actions';
import Loader from './../../components/Loader/Loader';

import './App.css';

export class App extends Component {

    constructor(props) {
        super(props);
        this.title = 'ORDERBOOK';

    }

    toggleModal() {
        this.showModal = !this.showModal;
        this.forceUpdate();
    }

    addOrder() {
        this.props.addOrder(this.selectCustomer.value);
        this.toggleModal();
    }
    render() {
        return (
            <div>
                {this.props.customers.length > 0 ?
                    <div>
                        < div className={this.showModal ? 'showModal modalBackDrop' : 'modalBackDrop'}
                            onClick={() => {
                                this.showModal = false;
                                this.forceUpdate(); // todo: ??? must be better way than this without keeping boolean in store ???
                                //this.setState({}) // triggers the same as forceUpdate...
                            }}>
                        </div>
                        <div className={this.showModal ? 'showModal modal' : 'modal'}>
                            <div id="modal" className="modal-content add-order-modal">
                                <div className="modal-header">
                                    Add order
                                </div>

                                <div className="modal-body">
                                    Select customer:

                                    <select id="select-customer" ref={(input) => this.selectCustomer = input} >
                                        {this.props.customers.map(c => {
                                            return <option key={c.id} value={c.id}>{c.name}</option>;
                                        })}
                                    </select>
                                </div>

                                <div className="modal-footer">
                                    <button id="cancel-button" onClick={() => this.toggleModal()}>cancel</button>
                                    <button id="ok-button" onClick={() => this.addOrder()} className="primary-button">ok</button>
                                </div>
                            </div>
                        </div>
                    </div>

                    : ''
                }
                <div className="App">


                    <header className="App-header">
                        <div className="halfCol header">
                            {this.title}
                        </div>
                        <div className="halfCol header ">
                            <button id="go-to-help-button" className="pullright header-button" onClick={() => this.props.history.push('/help')}>?</button>
                            <button id="add-order-button" className="pullright header-button" onClick={() => this.toggleModal()}>+</button>
                        </div>
                    </header>
                    {this.props.orders.length === 0 ? <Loader /> :
                        <div>
                            <ul className="list">
                                <li className="grid-header">
                                    <div className="quatreCol id-and-name">
                                        id
                                        <p className="username pullright">
                                            customer
                                        </p>
                                    </div>
                                    <div className="quatreCol textright">
                                        # items
                                    </div>
                                    <div className="quatreCol textright">
                                        total
                                    </div>
                                    <div className="quatreCol">
                                    </div>
                                </li>
                                <hr />
                                {this.props.orders.filter(o => !o.hasBeenPlaced).sort((a, b) => parseInt(a.id, 10) > parseInt(b.id, 10)).map(o => <List className="orderList" key={o.id} order={o} history={this.props.history} />)}



                                {this.props.orders.filter(o => o.hasBeenPlaced).length > 0 ?
                                    <li className="grid-header placed-orders-header">Placed Orders
                                        <hr />
                                    </li> : ''
                                }
                                {this.props.orders.filter(o => o.hasBeenPlaced).sort((a, b) => parseInt(a.id, 10) > parseInt(b.id, 10)).map(o => <List className="orderList placed-order" key={o.id} order={o} history={this.props.history} />)}

                            </ul>

                        </div>
                    }
                </div>
            </div >

        );
    }
}

const mapStateToProps = (state) => {
    return {
        orders: state.data.orders,
        customers: state.data.customers
    };

};


const mapDispatchToProps = (dispatch) => {
    return {
        addOrder: (customerId) => {
            dispatch(addOrder(customerId));
        }
    };
};

App.propTypes = {
    customers: PropTypes.array,
    orders: PropTypes.array,
    history: PropTypes.object,
    addOrder: PropTypes.func,
};


export default connect(mapStateToProps, mapDispatchToProps)(App);
