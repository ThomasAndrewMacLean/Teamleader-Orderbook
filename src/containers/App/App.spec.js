import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const initialState = {
    orders: {
        orders: [],
        products: [],
        users: []
    }
};

// {
//   'id': '1',
//   'customer-id': '1',
//   'items': [{
//       'product-id': 'B102',
//       'quantity': '10',
//       'unit-price': '4.99',
//       'total': '49.90'
//   }],
//   'total': '49.90'
// }

const mockStore = configureStore();
let wrapper;
let store;

beforeEach(() => {
    //creates the store with any initial state or middleware needed  
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}><App /></Provider>);
    //wrapper = shallow(<App store={store} />)
});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Provider store={store}><App /></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('title ORDERBOOK is renderd', () => {
    expect(wrapper.contains('ORDERBOOK')).toBe(true);
});

