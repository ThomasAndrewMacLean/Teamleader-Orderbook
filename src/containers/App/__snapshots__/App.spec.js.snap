// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div>
    <div
      className="modalBackDrop"
      onClick={[Function]}
    />
    <div
      className="modal"
    >
      <div
        className="modal-content add-order-modal"
        id="modal"
      >
        <div
          className="modal-header"
        >
          Add order
        </div>
        <div
          className="modal-body"
        >
          Select customer:
          <select
            id="select-customer"
          >
            <option
              value="1"
            >
              Coca Cola
            </option>
            <option
              value="2"
            >
              Teamleader
            </option>
            <option
              value="3"
            >
              Jeroen De Wit
            </option>
          </select>
        </div>
        <div
          className="modal-footer"
        >
          <button
            id="cancel-button"
            onClick={[Function]}
          >
            cancel
          </button>
          <button
            className="primary-button"
            id="ok-button"
            onClick={[Function]}
          >
            ok
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <div
        className="halfCol header"
      >
        ORDERBOOK
      </div>
      <div
        className="halfCol header "
      >
        <button
          className="pullright header-button"
          id="go-to-help-button"
          onClick={[Function]}
        >
          ?
        </button>
        <button
          className="pullright header-button"
          id="add-order-button"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </header>
    <div>
      <ul
        className="list"
      >
        <li
          className="grid-header"
        >
          <div
            className="quatreCol id-and-name"
          >
            id
            <p
              className="username pullright"
            >
              customer
            </p>
          </div>
          <div
            className="quatreCol textright"
          >
            # items
          </div>
          <div
            className="quatreCol textright"
          >
            total
          </div>
          <div
            className="quatreCol"
          />
        </li>
        <hr />
        <div
          className="Detail"
        >
          <li>
            <div
              className="quatreCol id-and-name"
            >
              1
              <p
                className="username pullright"
              >
                 
                Coca Cola
              </p>
            </div>
            <div
              className="quatreCol textright"
            >
              1
            </div>
            <div
              className="quatreCol textright"
            >
              €49.90
            </div>
            <div
              className="quatreCol"
            >
              <button
                className="pullright go-to-details-button"
                onClick={[Function]}
              >
                DETAILS
              </button>
            </div>
          </li>
        </div>
        <div
          className="Detail"
        >
          <li>
            <div
              className="quatreCol id-and-name"
            >
              2
              <p
                className="username pullright"
              >
                 
                Teamleader
              </p>
            </div>
            <div
              className="quatreCol textright"
            >
              1
            </div>
            <div
              className="quatreCol textright"
            >
              €24.95
            </div>
            <div
              className="quatreCol"
            >
              <button
                className="pullright go-to-details-button"
                onClick={[Function]}
              >
                DETAILS
              </button>
            </div>
          </li>
        </div>
        <div
          className="Detail"
        >
          <li>
            <div
              className="quatreCol id-and-name"
            >
              3
              <p
                className="username pullright"
              >
                 
                Jeroen De Wit
              </p>
            </div>
            <div
              className="quatreCol textright"
            >
              2
            </div>
            <div
              className="quatreCol textright"
            >
              €69.00
            </div>
            <div
              className="quatreCol"
            >
              <button
                className="pullright go-to-details-button"
                onClick={[Function]}
              >
                DETAILS
              </button>
            </div>
          </li>
        </div>
        
      </ul>
    </div>
  </div>
</div>
`;
